Darwin xz is broken in unstable

`xz.x86_64-darwin` is now failing, and blocking the unstable channel

https://hydra.nixos.org/build/145208114/nixlog/46

cc @NixOS/darwin-maintainers @thefloweringash

Minimal reproducer:
`nix-build pkgs/stdenv/darwin/make-bootstrap-tools.nix -A test-pkgs.stdenv.__bootPackages.stdenv.__bootPackages.stdenv.__bootPackages.xz`

Bisected to https://github.com/NixOS/nixpkgs/commit/4a2698221baa64754258e8a24706b4c1aeadf59a

Relevant snippet of config.log:
```
 165   │ configure:5938: checking for clang option to accept ISO C99
 166   │ configure:6087: clang  -c -g -O2  conftest.c >&5
 167   │ conftest.c:42:10: fatal error: 'stdbool.h' file not found
 168   │ #include <stdbool.h>
 169   │          ^~~~~~~~~~~
 170   │ 1 error generated.
 171   │ configure:6087: $? = 1
```

cc @toonn

--- a/pkgs/stdenv/darwin/default.nix
+++ b/pkgs/stdenv/darwin/default.nix
@@ -4,6 +4,7 @@
 , config
 , overlays
 , crossOverlays ? [ ]
+, bootstrapLlvmVersion ? if localSystem.isAarch64 then "11.1.0" else "7.1.0"
   # Allow passing in bootstrap files directly so we can test the stdenv bootstrap process when changing the bootstrap tools
 , bootstrapFiles ? if localSystem.isAarch64 then
     let
@@ -40,14 +41,11 @@ assert crossSystem == localSystem;
 let
   inherit (localSystem) system;
 
-  # Bootstrap version needs to be known to reference headers included in the bootstrap tools
-  bootstrapLlvmVersion = if localSystem.isAarch64 then "11.1.0" else "7.1.0";
-
   useAppleSDKLibs = localSystem.isAarch64;
   haveKRB5 = localSystem.isx86_64;
 
   # final toolchain is injected into llvmPackages_${finalLlvmVersion}
-  finalLlvmVersion = if localSystem.isAarch64 then "11" else "7";
+  finalLlvmVersion = lib.versions.major bootstrapLlvmVersion;
   finalLlvmPackages = "llvmPackages_${finalLlvmVersion}";
 
   commonImpureHostDeps = [

--- a/pkgs/stdenv/darwin/make-bootstrap-tools.nix
+++ b/pkgs/stdenv/darwin/make-bootstrap-tools.nix
@@ -254,6 +254,8 @@ in rec {
     '';
   };
 
+  bootstrapLlvmVersion = llvmPackages.llvm.version;
+
   bootstrapFiles = {
     sh      = "${build}/on-server/sh";
     bzip2   = "${build}/on-server/bzip2";
@@ -408,7 +410,7 @@ in rec {
     system = if crossSystem != null then crossSystem else system;
 
     stdenvStages = args: let
-        args' = args // { inherit bootstrapFiles; };
+        args' = args // { inherit bootstrapLlvmVersion bootstrapFiles; };
       in (import (test-pkgspath + "/pkgs/stdenv/darwin") args').stagesDarwin;
   };
 }

--- a/pkgs/top-level/release.nix
+++ b/pkgs/top-level/release.nix
@@ -174,7 +174,8 @@ let
               # Lightweight distribution and test
               inherit (bootstrap) dist test;
               # Test a full stdenv bootstrap from the bootstrap tools definition
-              inherit (bootstrap.test-pkgs) stdenv;
+              # TODO re-enable with https://github.com/NixOS/nixpkgs/pull/126411
+              # inherit (bootstrap.test-pkgs) stdenv;
             };
 
           # Cross compiled bootstrap tools

